app/javascript/components/supplemental_application/SupplementalApplicationPage.js

```diff
@@ -33,17 +33,19 @@ const SupplementalApplicationPage = ({ application, statusHistory, formFields, o
         formFields={formFields}
         onSubmit={onSubmit}
         fileBaseUrl={fileBaseUrl}
+        units={units}
       />
     </CardLayout>
   )
 }
 
-const mapProperties = ({application, statusHistory, file_base_url}) => {
+const mapProperties = ({application, statusHistory, file_base_url, units}) => {
   return {
     application: mapApplication(application),
     statusHistory: mapList(mapFieldUpdateComment,statusHistory),
     onSubmit:  (values) => updateApplicationAction(values),
-    fileBaseUrl: file_base_url
+    fileBaseUrl: file_base_url,
+    units: units.map(unit => mapUnit(unit))
```

Comment:
I would use `mapList` or `map` from lodash to avoid null pointer as a best practice.

Look in this same shape how mapList is being used